softwareComponents/rofiHalInc/include/rofi_hal.hpp

```diff
@@ -622,14 +622,23 @@ class RoFI {
     void reboot() { return _impl->reboot(); }
 
     /**
-     * \brief Call callback after given delay.
+     * \brief Call callback after given delay. The call is blocking.
      *
      * You can assume that the callback will always be called exactly once.
      * \param ms delay in milliseconds
      * \param callback non-empty callback to be called after the delay
      */
     static void wait( int ms, std::function< void() > callback );
 
+    /**
+     * \brief Call callback after given delay. The call is non-blocking.
+     *
+     * You can assume that the callback will always be called exactly once.
+     * \param ms delay in milliseconds
+     * \param callback non-empty callback to be called after the delay
+     */
+    static void delay( int ms, std::function< void() > callback );
```

Comment:
The naming and even the contract is very confusing. In my opinion, we should have:
- `void delay( int ms )` that takes no function and simply blocks the caller for a given period of time.
- `void postpone( int ms, std::function< void() > callback )` that returns immediately and calls `callback` after given period of time in unspecified context.

@jajik: What do you think?